### PR TITLE
core: expand documentation of thread_create()

### PR DESCRIPTION
**Pull request by authmillenon**
Adds specific documentation for the error return values of `thread_create()`.

### CODE REVIEW

File: core/include/thread.h [hunk 1]
```diff
@@ -76,8 +76,10 @@
  * @param[in] arg       the argument to the function
  * @param[in] name      a human readable descriptor for the thread
  *
- * @return              value ``<0`` on error
- * @return              pid of newly created task, otherwise
+ * @return              PID of newly created task, on success
```
Comment (LudwigOrtmann) [inline core/include/thread.h]: remove comma

**Comment by authmillenon** [comment]
Addressed comment

**Comment by LudwigOrtmann** [comment]
squash -> ACK

**Comment by authmillenon** [comment]
Done. Does it need a second ACK since this is technically a core change?

**Comment by OlegHahm** [comment]
I don't think so - I ack it anyway.

File: core/include/thread.h [hunk 2]
```diff
@@ -76,8 +76,10 @@
  * @param[in] arg       the argument to the function
  * @param[in] name      a human readable descriptor for the thread
  *
- * @return              value ``<0`` on error
- * @return              pid of newly created task, otherwise
+ * @return              PID of newly created task on success
+ * @return              -EINVAL, if @p priority is higher or equal
```
Comment (gebart) [inline core/include/thread.h]: "greater than or equal to"

Comment (authmillenon) [inline core/include/thread.h]: Guess I was "high" there :grin:

**Comment by authmillenon** [comment]
Squashed address to comment of @gebart already

**Pull request merged**